frontend/src/screens/GameScreen/gameScreen.tsx

```diff
@@ -0,0 +1,247 @@
+            }
+            return (<DraxView payload={word} key={word}
+                style={styles.draxView}
+                draggingStyle={{ opacity: 0.3 }}
+                dragReleasedStyle={{ opacity: 0.3 }}
+                longPressDelay={100}>
+                <Text style={styles.english}>{word}</Text>
+            </DraxView>)
+        })
+    }
+
+    const wordDroppedFunction = (turkishWords: MaybeWordPair[], word: MaybeWordPair, i : number) => {
+      return (newWord : string) => {
+      const newTurkishWords = turkishWords.map(({ english, turkish, correctEnglishWord }) => {
+            if (english === newWord) {
+                return { turkish, correctEnglishWord }
+            } else {
+                return { turkish, correctEnglishWord, english }
+            }
+        })
+        newTurkishWords[i] = { english: newWord, turkish: word.turkish, correctEnglishWord: word.correctEnglishWord }
+        setTurkishWords(newTurkishWords)
+      }
+    }
+
+    const removeWordFunction = (turkishWords: MaybeWordPair[], word: MaybeWordPair, i: number) => {
+      return () => {
+        const newTurkishWords = [...turkishWords]
+        newTurkishWords[i] = { turkish: word.turkish, correctEnglishWord: word.correctEnglishWord }
+        setTurkishWords(newTurkishWords)
+    }
+    }
+
+    const renderTurkishWords = () => {
+        return turkishWords?.map((word, i) => (
+            <DroppableRow
+                key={word.turkish}
+                showingResults={submitted}
+                wordDropped={wordDroppedFunction(turkishWords, word, i)}
+                turkish={word.turkish}
+                english={word.english}
+                correctEnglish={word.correctEnglishWord}
+                removeWord={removeWordFunction(turkishWords, word, i)}
+                isPairing={props.isPairing}
+            />))
+    }
+
+    const renderSubmittedButtons = () => {
+        return (<View style={styles.doneContainer}><PrimaryButton
+            disabled={isLoading}
+            onPress={() => {
+                setIsLoading(true);
+                restartRound()
+            }}
+            text="play again" />
+            <SecondaryButton
+                disabled={isLoading}
+                text="end session"
+                onPress={() => {
+                    fi.endRound(currentRoundId);
+                    fi.endSession(sessionId);
+                    navigation.navigate("HomeScreen");
+                }} />
+        </View>)
+    }
+
+    const renderInProgressButtons = () => {
+        return (<View style={styles.doneContainer}>
+            <PrimaryButton
+                disabled={!canClickDoneButton || isLoading}
+                text="done"
+                onPress={() => setSubmitted(true)} />
+        </View>
+        )
+    }
+
+    const correctValues = turkishWords?.filter(({ english, correctEnglishWord }) => english === correctEnglishWord).length ?? 0;
```

Comment:
Also nitpicky, but what do you think of renaming this `numCorrectPairs` or something similar? I feel like `correctValues` implies a list of values, whereas this is just meant to be an int.